core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -73,6 +74,10 @@ private[spark] class AppStatusListener(
   // around liveExecutors.
   @volatile private var activeExecutorCount = 0
 
+  private val inputDataSetId = new AtomicLong(0)
+  private val outputDataSetId = new AtomicLong(0)
+  private val maxRecords = conf.getInt("spark.data.maxRecords", 1000)
```

Comment:
What's this `spark.data.maxRecords` for? Maybe you should follow the config in core/src/main/scala/org/apache/spark/status/config.scala